kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/format/KafkaEntryFormatterHeader.java

```diff
@@ -34,6 +34,19 @@
         return messageMetadata;
     }
 
+    public PulsarApi.MessageMetadata getMessageMetadataWithNumberMessages(int numMessages) {
+        final PulsarApi.MessageMetadata.Builder builder = PulsarApi.MessageMetadata.newBuilder();
+        builder.addProperties(PulsarApi.KeyValue.newBuilder()
+                .setKey("entry.format")
+                .setValue(EntryFormatterFactory.EntryFormat.KAFKA.name().toLowerCase())
+                .build());
+        builder.setProducerName("");
+        builder.setSequenceId(0L);
+        builder.setPublishTime(0L);
+        builder.setNumMessagesInBatch(numMessages);
+        return builder.build();
+    }
+
```

Comment:
`KafkaEntryFormatterHeader` is a singleton class because we only need to initialize once for the header that is just used to indicate the entry format is kafka.

Is `setNumMessagesInBatch` necessary here? If yes, the other code of `KafkaEntryFormatterHeader` could be removed.

Reply:
Yes，I think `setNumMessagesInBatch` is nessacery here since we need to caculate the start offset in an entry when handling fetch request and now the an entry only has the end offset.

Reply:
OK, got it.